src/topology/bounded_continuous_function.lean

```diff
@@ -495,4 +498,101 @@ instance : normed_space 𝕜 (α →ᵇ β) := ⟨λ c f, norm_of_normed_group_l
 
 end normed_space
 
+section normed_ring
+/-!
+### Normed ring structure
+
+In this section, if `R` is a normed ring, then we show that the space of bounded
+continuous functions from `α` to `R` inherits a normed ring structure, by using
+pointwise operations and checking that they are compatible with the uniform distance. -/
+
+variables [topological_space α] {R : Type*} [normed_ring R]
+
+instance : ring (α →ᵇ R) :=
+{ one := const α 1,
+  mul := λ f g, of_normed_group (f * g) (f.2.1.mul g.2.1) (∥f∥ * ∥g∥) $ λ x,
+    le_trans (normed_ring.norm_mul (f x) (g x)) $
+      mul_le_mul (f.norm_coe_le_norm x) (g.norm_coe_le_norm x) (norm_nonneg _) (norm_nonneg _),
+  one_mul := λ f, ext $ λ x, one_mul (f x),
+  mul_one := λ f, ext $ λ x, mul_one (f x),
+  mul_assoc := λ f₁ f₂ f₃, ext $ λ x, mul_assoc _ _ _,
+  left_distrib := λ f₁ f₂ f₃, ext $ λ x, left_distrib _ _ _,
+  right_distrib := λ f₁ f₂ f₃, ext $ λ x, right_distrib _ _ _,
+  .. bounded_continuous_function.add_comm_group }
+
+instance : normed_ring (α →ᵇ R) :=
+{ norm_mul := λ f g, norm_of_normed_group_le _ (mul_nonneg (norm_nonneg _) (norm_nonneg _)) _,
+  .. bounded_continuous_function.normed_group }
+
+end normed_ring
+
+section normed_algebra
+/-!
+### Normed algebra structure
+
+In this section, if `γ` is a normed algebra, then we show that the space of bounded
+continuous functions from `α` to `γ` inherits a normed algebra structure, by using
+pointwise operations and checking that they are compatible with the uniform distance. -/
+
+variables {𝕜 : Type*} [nondiscrete_normed_field 𝕜]
+variables [topological_space α] [normed_group β] [normed_space 𝕜 β]
+variables [normed_ring γ] [normed_algebra 𝕜 γ]
+variables {f g : α →ᵇ γ} {x : α} {c : 𝕜}
+
+/-- `bounded_continuous_function.const` as a `ring_hom`. -/
+def C : 𝕜 →+* (α →ᵇ γ) :=
+{ to_fun    := λ (c : 𝕜), const α ((algebra_map 𝕜 γ) c),
+  map_one'  := by rw (algebra_map 𝕜 γ).map_one; exact rfl,
```

Comment:
```suggestion
  map_one'  := by { rw (algebra_map 𝕜 γ).map_one, refl },
```
`exact rfl` should in general be replaced by `refl`, which is a tactic doing (essentially) the same thing. Also, we use `;` only when the first goal spans several subgoals that can be solved in the same way. Otherwise, use the comma: it makes it easier to see the evolution of the goal by moving the cursor. Same thing for the next 3 lines.

Reply:
(Note that in 3.13 we should be able to see the goals created by `;` when we move the cursor.)

Reply:
That's great news!

Reply:
Thanks, this is useful.  Actually I just rewrote to match the syntax of [earlier instances](https://github.com/leanprover-community/mathlib/blob/d20168154cbcf4cb1dfca347943ff1a82301a509/src/topology/bounded_continuous_function.lean#L516), which has no reflexivity conclusion (and no semicolons).